tests/unit/test_utils.py

```diff
@@ -802,20 +802,6 @@ def test_basic(self) -> None:
         assert hidden.redacted == "######"
         assert hidden.secret == "my-secret"
 
-    def test_equality_with_str(self) -> None:
```

Comment:
Hmm... I think this test should have been preserved.

Reply:
The comparisons in this test are no longer type-safe which is why I removed it. I can adjust the class to support comparisons with strings if that's wanted.

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

tests/unit/test_utils.py:813: error: Non-overlapping equality check (left
operand type: "HiddenText", right operand type: "str")  [comparison-overlap]
            assert hidden != hidden.secret
                   ^~~~~~~~~~~~~~~~~~~~~~~
tests/unit/test_utils.py:814: error: Non-overlapping equality check (left
operand type: "str", right operand type: "HiddenText")  [comparison-overlap]
            assert hidden.secret != hidden
                   ^~~~~~~~~~~~~~~~~~~~~~~
tests/unit/test_utils.py:816: error: Non-overlapping equality check (left
operand type: "HiddenText", right operand type: "str")  [comparison-overlap]
            assert hidden != hidden.redacted
                   ^~~~~~~~~~~~~~~~~~~~~~~~~
tests/unit/test_utils.py:817: error: Non-overlapping equality check (left
operand type: "str", right operand type: "HiddenText")  [comparison-overlap]
            assert hidden.redacted != hidden
                   ^~~~~~~~~~~~~~~~~~~~~~~~~
Found 4 errors in 1 file (checked 292 source files)
```

Reply:
That's because you removed the `__eq__` comparison from the `HiddenText` class...

Reply:
It doesn't seem like the class was meant to be compared with strings in the first place (other than in tests) so it's debatable whether it's worth keeping it, but I've rolled this back in https://github.com/pypa/pip/pull/12571/commits/bde667d76513de97e0c6d22a89ad96d7a947a273 anyway. 
https://github.com/pypa/pip/blob/31437d675abfa1563cffb3f68a0a23f3cbd68afa/src/pip/_internal/utils/misc.py#L594-L601